### DISCUSSION TITLE
Bug - Missing String representation of the fastapi.HTTPException when initializing with parameter names

### DISCUSSION THREAD

**Discussion by CribberSix** [fastapi/fastapi#6064] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi 
import pytest

# When creating the function with parameter names, the string representation does not work:
exc = fastapi.HTTPException(status_code=401, detail="Authentication error: Authorization header is missing.")
print(f"String representation of the exception: '{exc}'")

# When creating the function without parameter names, the string representation works:
exc = fastapi.HTTPException(401, "Authentication error: Authorization header is missing.")
print(f"String representation of the exception without params: '{exc}'")


# The missing string representation becomes a problem when using pytest to check for the raised
# exception as well as the exception message, as the match="..." parameter value is compared to the 
# string representation of the exception, which does not exist when the exception is initialized
# with parameter names. 
def test_for_invalid_something(): 
    with pytest.raises(fastapi.HTTPException, match=r"Authentication error: Authorization header is missing\."):        
        raise fastapi.HTTPException(status_code=401, detail="Authentication error: Authorization header is missing.")
```

### Description

As the code shows, when we initialize the exception with parameter names, the string representation is empty. I'm not sure what the cause could be... 

### Wanted Solution

When creating the function with parameter names, the string representation should also work-

### Wanted Code

```python
exc = fastapi.HTTPException(status_code=401, detail="Authentication error: Authorization header is missing.")
print(f"String representation of the exception: '{exc}'")
```

### Operating System

Windows

### FastAPI Version

0.89.1

### Python Version

3.8.10

**Comment by Kludex** [comment, marked as answer]
This is solved for some time now.
